File: listen.py
import json
import requests
import base64
import sounddevice as sd
from scipy.io.wavfile import write
from io import BytesIO

duration = 6  # seconds
fs = 44100

recorded_file_name = "output.mp3"

# records song and saves to file. easier to save then open, and it overall doesn't take too long.
def record():
    print('Recording...')
    recorded = sd.rec(int(duration * fs), samplerate=fs, channels=1)
    sd.wait()

    write(recorded_file_name, fs, recorded) 

    print('Done Recording')

# recognize the current song playing. also records
def recognize():
    record()
    sound_file = open('./'+recorded_file_name,"rb")
    sound_data_binary = sound_file.read()
    sound_data = (base64.b64encode(sound_data_binary))

    api_token_file = open('key', 'r')

    data = {
        'return': 'spotify',
        'api_token': api_token_file.read().split("\n")[0],
        'audio': sound_data
    }

    result = requests.post('https://api.audd.io/', data=data)

    response = json.loads(result.text)

    try:
        print(response['result']['title'])
        # print(response['result']['artist'])
        imgurl = response['result']['spotify']['album']['images'][0]['url']
    except:
        print('### error 1 ###')
        # print(response)
        imgurl = 'https://i.ibb.co/Q85tMW2/e1.png' # error 1

    print(imgurl)
    return imgurl

recognize()

File: listen.js
var request = require("request");
const fs = require('fs');
const path = require('path');
var http = require('http');

function getData() {
  return new Promise((res, rej) => {
    fs.readFile(path.join(__dirname, 'samples/newslaves.m4a'), 'base64', (err, resp) => {
      console.log(resp);
      var data = {
        'return': 'spotify',
        'api_token': '0295a1c0139a030849dd81359d92122a',
        'audio': resp,
      };

      request({
        uri: 'https://api.audd.io/',
        form: data,
        method: 'POST'
      }, function (err, resp, body) {
        let data = JSON.parse(body);

        res(data);
      });
    });
  })

}

http.createServer(function (req, res) {
  getData().then( d => {
    let c = JSON.stringify(d);
    console.log(c);
    res.writeHead(200, {'Content-Type': 'text/html'});
    res.write(`
    <h1>${d.result.title}</h1>
    <h2>${d.result.artist}</h2>
    <h2>${d.result.album}</h2>
    `);
    res.write(`
      <img width='200' height='200' src="${d.result.spotify.album.images[0].url}">
      <br>
      <p style='margin-top:200px;'>${c}</p>
    `);

    // res.write(c + '\n')
    res.end(); //end the response
  })
}).listen(8080); //the server object listens on port 8080